### ISSUE TITLE
Segmentation Fault on Mastodon installation

### ISSUE THREAD

**Issue by motorailgun** [oracle/truffleruby#2801]
### Guest (setting up mastodon):
```
Ubuntu 22.04.1 LTS
truffleruby 22.3.0, like ruby 3.0.3, GraalVM CE Native [x86_64-linux]
rbenv master@6717c62dc18a81e2d3ce3e3d6dcf70839dd981c9
Mastodon main@v4.0.2

4core/4GiB, KVM
```
### Host:
```
Arch Linux/Kernel 6.0.10-zen2-1-zen
libvirt 1:8.9.0-1
virt-manager 4.1.0-1
qemu-common 7.1.0-10

Ryzen5 3600, 64GiB
```

## Synopsis
TruffleRuby crashes with SegFault on Mastodon setup. Both native and graalVM affected. Log is attached on last.

## Steps to reproduce
Following [the Mastodon setup guide](https://docs.joinmastodon.org/admin/install/), crash occurs on `Generating a configuration` step, right after pressing enter on Redis password. However there's diff to use TruffleRuby:  

### on 'installing Ruby' step:
for reproducibility, at '_And proceed to install rbenv and rbenv-build:_'
```diff
- cd ~/.rbenv && src/configure && make -C src
+ cd ~/.rbenv && git checkout 6717c62dc18a81e2d3ce3e3d6dcf70839dd981c9 && src/configure && make -C src
```

at '_Once this is done, we can install the correct Ruby version:_'
```diff
- RUBY_CONFIGURE_OPTS=--with-jemalloc rbenv install 3.0.4
- rbenv global 3.0.4
+ rbenv install truffleruby+graalvm-22.3.0
+ rbenv global truffleruby+graalvm-22.3.0
```
Note: both truffleruby-22.3.0 and truffleruby+graalvm-22.3.0 have same problem

### on 'Checking out the code' step:
```diff
- git checkout $(git tag -l | grep -v 'rc[0-9]*$' | sort -V | tail -n 1)
+ git checkout v4.0.2
```

### on 'Installing the last dependencies' step:
at '_Now to install Ruby and JavaScript dependencies:_', __before everything__,
```diff
+ echo 'truffleruby+graalvm-22.3.0' > ./.ruby-version
```

and then proceed. On `RAILS_ENV=production bundle exec rake mastodon:setup`, put `localhost` to Domain name, and change `Are you using Docker to run Mastodon?` to no. Everyting else is default or blank.

The crash will occur right after pressing enter on Redis password prompt.

## Log
[segfault.log](https://github.com/oracle/truffleruby/files/10135524/segfault.log)

This is just my personal opinion, but Mastodon might be a good test for TruffleRuby compatibility.

**Comment by andrykonchin** [comment]
Thank you for the report.

**Comment by eregon** [comment]
Given it's around loading/using Redis (hiredis specifically), I suspect it's fixed by https://github.com/oracle/truffleruby/commit/3877eeb6d9016d796f1fbc385c908c781e870517 (on master but not yet in a release). I'll verify that.

**Comment by motorailgun** [comment]
> Given it's around loading/using Redis (hiredis specifically), I suspect it's fixed by [3877eeb](https://github.com/oracle/truffleruby/commit/3877eeb6d9016d796f1fbc385c908c781e870517) (on master but not yet in a release). I'll verify that.

Looks like working with the current truffleruby-dev! Thank you :heart:
